### PR TITLE
[posix] clear SPI tx buffer after usage

### PR DESCRIPTION
**Pull request by hastigondaliya**
When ot-daemon is serving an interrupt from the RCP and has no data to transmit, it still sends the same data as the previous transmit with data En=0.

This commits clears "mSpiTxPayloadSize" buffer once the packet has been successfully transmitted.

### CODE REVIEW

**Comment by zhanglongxia** [comment]
> When ot-daemon is serving an interrupt from the RCP and has no data to transmit, it still sends the same data as the previous transmit with data En=0.
> 

I think if there is no data to transmit, the SPI master will transmit the [mEmptySendFrameZeroAccept or mEmptySendFrameFullAccept](https://github.com/openthread/openthread/blob/main/src/ncp/ncp_spi.cpp#L206). It won't transmit the same data as the previous transmit.

File: src/posix/platform/spi_interface.cpp [hunk 1]
```diff
@@ -569,6 +569,8 @@ otError SpiInterface::PushPullSpi(void)
             mInterfaceMetrics.mTxFrameCount++;
             mInterfaceMetrics.mTxFrameByteCount += mSpiTxPayloadSize;
 
+            // Clear tx buffer after usage
+            memset(&mSpiTxFrameBuffer[kSpiFrameHeaderSize], 0, mSpiTxPayloadSize);
```
Comment (abtink) [inline src/posix/platform/spi_interface.cpp:573]: We should not need this though it is most likely harmless.

The `mSpiTxPayloadSize` is set to zero and `mSpiTxIsRead` to `false` which indicates there is no data to tx.

When we prepare the SPI frame in a future trasnisation in `PushPullSpi()` we set `txFrame.SetHeaderDataLen(0)` which set the "data len" field to zero  and peripheral would then ignore any clocked data bytes.

Comment (hastigondaliya) [reply src/posix/platform/spi_interface.cpp:573]: Hello @abtink,

There is no functional problem. This change is more conceptive when debugging.

The `mSpiTxFrameBuffer` is cleared for debugging purposes. After a successful transmission, the `data len` is set to zero, but the data itself remains unchanged. As a result, in debug logs, we may still see the same data that was previously sent, even if the `data len` is "zero". 

To facilitate debugging, the data buffer is cleared for convenience.

Comment (abtink) [reply src/posix/platform/spi_interface.cpp:573]: Sounds good to me. Thanks.

**Review by abtink — APPROVED** [review]
👍

File: src/posix/platform/spi_interface.cpp (hunk 1 above)
Comment (jwhui) [reply src/posix/platform/spi_interface.cpp:573]: This sounds like an issue with the logging. Is there value to printing any bytes when `mSpiTxPayloadSize` is set to zero? Ideally, we are not wasting cycles doing something that is generally unnecessary.

Comment (hastigondaliya) [reply src/posix/platform/spi_interface.cpp:573]: @zhanglongxia, Could you please help to review this?

Comment (hastigondaliya) [reply src/posix/platform/spi_interface.cpp:573]: Hi @zhanglongxia,

This is a gentle reminder to review this PR. Can you please have a look at this?

Comment (zhanglongxia) [reply src/posix/platform/spi_interface.cpp:573]: When there is no data to be transmitted to the RCP, the [SPI header data length](https://github.com/openthread/openthread/blob/main/src/posix/platform/spi_interface.cpp#L408) is set to `0` and `mSpiTxPayloadSize ` is `0`. The length of the TX SPI frame should be `sizeof(SpiHeader)`. But if RCP has data to transmit to the host at this time, the actual TX SPI frame length is larger than the `sizeof(SpiHeader)`, the data payload filed of the TX SPI frame uses the previous transmitted data payload in the `&mSpiTxFrameBuffer[kSpiFrameHeaderSize]`. The [log function](https://github.com/openthread/openthread/blob/main/src/posix/platform/spi_interface.cpp#L474) shows the actual TX SPI frame.

This case won’t affect the working of SPI. I think it is useful to reset the TX buffer to make the log clearer when debugging the SPI interface. The `spi_interface.cpp` prints out the raw TX SPI frame using the `DEBUG` level, the max SPI frame is set to [8192 + 16](https://github.com/openthread/openthread/blob/main/src/posix/platform/spi_interface.hpp#L244) by default and resetting the buffer may take much time. How about changing the code like this？

```c++
if (otLoggingGetLevel() >= OT_LOG_LEVEL_DEBG) {
  memset(&mSpiTxFrameBuffer[kSpiFrameHeaderSize], 0, mSpiTxPayloadSize);
}
```

Comment (abtink) [reply src/posix/platform/spi_interface.cpp:573]: I think the idea to help with logging on the other end (other device), so checking `otLoggingGetLevel() >= OT_LOG_LEVEL_DEBG` on sender may not matter?

Comment (zhanglongxia) [reply src/posix/platform/spi_interface.cpp:573]: If that is the case, it should be fine to not check `otLoggingGetLevel() >= OT_LOG_LEVEL_DEBG`.

**Review by zhanglongxia — APPROVED** [review]
LGTM

File: src/posix/platform/spi_interface.cpp (hunk 1 above)
Comment (hastigondaliya) [reply src/posix/platform/spi_interface.cpp:573]: Hi,
Could you please help to merge these changes if this looks okay?

**Review by jwhui — APPROVED** [review]

**Pull request merged**